src/main/java/com/app/bookstore/dto/CreateBookRequestDto.java

```diff
@@ -1,13 +1,24 @@
 package com.app.bookstore.dto;
 
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Pattern;
+import jakarta.validation.constraints.Positive;
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
+    @Size(min = 4, max = 20, message = "length should be between 4 and 20")

```

Comment:
IMO you specified too strict validation rules. You can set the same maximum length as the maximum field length in DB. Regarding minimal length, I think there are some books with shorter names than 4 symbols.